.github/workflows/r_nightly.yml

```diff
@@ -65,54 +68,58 @@ jobs:
           fi
           echo $PREFIX
 
-          archery crossbow download-artifacts -f r-nightly-packages -t binaries --skip-pattern-validation $PREFIX
+          archery crossbow download-artifacts -f r-nightly-packages -t binaries $PREFIX
+
+          if [ -n "$(ls -A binaries/*/*/)" ]; then
+            echo "Found files!"
+            echo "GOT_FILES=true" >> $GITHUB_ENV
+          else
+            echo "No files found. Stopping upload."
+            echo "GOT_FILES=false" >> $GITHUB_ENV
+          fi
       - name: Build Repository
+        if: env.GOT_FILES == 'true'
         shell: Rscript {0}
         run: |
+          repo_root <- "repo"

```

Comment:
Can you please liberally add code comments in this script to explain what you're doing and why? Particularly useful might be to describe/give examples of the inputs and outputs; otherwise it's not clear why all of the regex-ing.